### PR TITLE
Add more examples to storage documentation.

### PR DESCRIPTION
**Pull request by yhontyk**
Adds more information about storage in odo - how to add it, check its state, and delete it.
Also removing an outdated command from the CLI reference.

Branch 4.5 and 4.6

### CODE REVIEW

File: modules/developer-cli-odo-adding-storage-to-the-application-components.adoc [hunk 1]
```diff
@@ -5,15 +5,67 @@
 [id="adding-storage-to-the-application-components_{context}"]
 = Adding storage to the application components
 
-Persistent storage keeps data available between restarts of {odo-title}. You can add storage to your components with the `odo storage` command.
+Persistent storage keeps data available between restarts of {odo-title}.
+Use `odo storage` to add data that must persist, such database files, dependencies, build artifacts (for example, a `.m2` directory for Maven applications), to your application.
```
Comment (bobfuru) [inline modules/developer-cli-odo-adding-storage-to-the-application-components.adoc:9]: This sentence needs some work and is missing an "as". Suggest something like:
Use the `odo storage` command to add persistent data to your application. Examples of data that must persist include database files, dependencies, and build artifacts, such as a Maven directory.

File: modules/developer-cli-odo-adding-storage-to-the-application-components.adoc [hunk 2]
```diff
@@ -5,15 +5,67 @@
...
+$ odo storage list
+----
++
+.Example output
++
+[source,terminal]
+----
+The component 'nodejs' has the following storage attached:
+NAME           SIZE     PATH
+mystorage      1Gi      /data     Locally Deleted
```
Comment (bobfuru) [inline modules/developer-cli-odo-adding-storage-to-the-application-components.adoc:63]: Should there be a `STATE` header in above `Locally Deleted` or does it not report that? I'm not sure.

**Review by bobfuru — COMMENTED** [review]
A suggestion for read wording the new sentence in the intro and a question. Let me know when ready for another look. Also, I see Vikram is a requested reviewer...do you want his ack before merging?

File: modules/developer-cli-odo-adding-storage-to-the-application-components.adoc (hunk 1 above)
Comment (boczkowska) [reply modules/developer-cli-odo-adding-storage-to-the-application-components.adoc]: @bobfuru thank you! I've been struggling for a while with this sentence. Changed.

File: modules/developer-cli-odo-adding-storage-to-the-application-components.adoc (hunk 2 above)
Comment (boczkowska) [reply modules/developer-cli-odo-adding-storage-to-the-application-components.adoc:63]: @bobfuru you're right! There i a header: https://github.com/openshift/odo/pull/2584
Updated.

**Comment by boczkowska** [comment]
@bobfuru thank you for the review! I've addressed your comments.

**Comment by bobfuru** [comment]
> @bobfuru thank you for the review! I've addressed your comments.

You're welcome! Would you like this merged and CP'ed to 4.5 and 4.6?

File: modules/developer-cli-odo-adding-storage-to-the-application-components.adoc [hunk 3]
```diff
@@ -5,15 +5,67 @@
...
 
 .Procedure
 
-* Add storage to your components:
+. Add the storage to your component:
 +
 [source,terminal]
 ----
-$ odo storage create nodestorage --path=/opt/app-root/src/storage/ --size=1Gi
+$ odo storage create <storage-name> --path=<path-to-the-directory> --size=<size>
```
Comment (bobfuru) [inline modules/developer-cli-odo-adding-storage-to-the-application-components.adoc:17]: s/<storage-name>/<storage_name>/
s/<path-to-the-directory>/<path_to_the_directory>/

Comment (boczkowska) [reply modules/developer-cli-odo-adding-storage-to-the-application-components.adoc]: @bobfuru my apologies! Somehow I missed the notification about your comment. 
Fixed.

Comment (bobfuru) [reply modules/developer-cli-odo-adding-storage-to-the-application-components.adoc]: @boczkowska - LGTM, let me know if you're ready for merge/CP

Comment (boczkowska) [reply modules/developer-cli-odo-adding-storage-to-the-application-components.adoc]: @bobfuru it is ready to be merged and cherry-picked. Could you please do it? I don\t have the merging rights.
Thank you.

**Comment by bobfuru** [comment]
All set with merge and CP to 4.5 and 4.6, @boczkowska

**Pull request merged**